src/data/fintype/basic.lean

```diff
@@ -13,11 +11,62 @@ import data.array.lemmas
 import order.well_founded
 import group_theory.perm.basic
 
+/-!
+# Finite types
+
+This file defines a typeclass to state that a type is finite.
+
+## Main declarations
+
+* `fintype α`:  Typeclass saying that a type is finite. It takes as fields a `finset` and a proof
+  that all terms of type `α` are in it.
+* `finset.univ`: The finset of all elements of a fintype.
+* `fintype.card α`: Cardinality of a fintype. Equal to `finset.univ.card`.
+* `perms_of_finset s`: The finset of permutations of the finset `s`.
+* `fintype.trunc_equiv_fin`: A fintype `α` is computably equivalent to `fin (card α)`. The
+  `trunc`-free, noncomputable version is `fintype.equiv_fin`.
+* `fintype.trunc_equiv_of_card_eq` `fintype.equiv_of_card_eq`: Two fintypes of same cardinality are
+  equivalent. See above.
+* `fin.equiv_iff_eq`: `fin m ≃ fin n` iff `m = n`.
+* `infinite α`: Typeclass saying that a type is infinite. Defined as `fintype α → false`.
+* `not_fintype`: No `fintype` is infinite.
```

Comment:
I find `not_fintype` a bit weird, but can't really find a way to clarify it via the docstring.
```suggestion
* `not_fintype`: No `fintype` has an `infinite` instance.
```